packages/core-contracts/contracts/utils/Ownable.sol

```diff
@@ -0,0 +1,37 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+abstract contract Ownable {
+    address public owner;
+    address public nominatedOwner;
```

Comment:
This is a very interesting dilemma that we first bump into with this PR:

This Ownable implementation is perfect for an architecture like Synthetix v2x, but incompatible with the v3 architecture.

I.e. we wouldn't be able to use this in a module in the v3 architecture. The deployer's AST parser would point out the invalid use of storage. See `packages/deployer/test/sample-project/contracts/modules/OwnerModule.sol`.

So, we need to make a decision. Do we make our core-contracts library compatible with the architecture? If so, should we abstract away anything that writes a value to storage?